prime-router/src/main/kotlin/cli/tests/Hl7Tests.kt

```diff
@@ -57,4 +57,138 @@ class Hl7Ingest : CoolTest() {
+        """.trimIndent()
+        val badHl7Pairs = listOf(
+            /* Missing Required Fields */
+            Pair("MISSING PATIENT LASTNAME", cleanedHl7Message.replace("Patlast", "")),
+            Pair("MISSING ORDERING FACILITY STATE", cleanedHl7Message.replace("^IG^", "")),
+            Pair("MISSING MESSAGE ID", cleanedHl7Message.replace("3719999", "")),
+            Pair("MISSING TESTING LAB CLIA", cleanedHl7Message.replace("10D08761999", "")),
+            Pair("MISSING PATIENT STATE", cleanedHl7Message.replace("^MD^", "^^")),
+            /* Invalid data type */
+            Pair("INVALID DOB Partial", cleanedHl7Message.replace("19670202", "19")),
+            Pair("INVALID DOB Special Chars", cleanedHl7Message.replace("19670202", special_char)),
+            /* Bad hl7 files */
+            Pair("EMPTY FILE", ""),
+            Pair("CSV FILE", csv),
+            Pair("XML FILE", xml_data),
+            Pair("JSON", "{\"alive\": true}"),
+            /* Bad data */
+            Pair("Duplicate PID", cleanedHl7Message + "PID|||||Richards^Mary||19340428|F|||||||||||||||||||\\nPID|||||||19700510105000|M|||||||||||||||||||"),
+            Pair("Only MSH", "MSH|^~\\&|CDC PRIME - Atlanta, Georgia (Dekalb)^2.16.840.1.114222.4.1.237821^ISO|Avante at Ormond Beach^10D0876999^CLIA|PRIME_DOH|Prime Data Hub|20210210170737||ORU^R01^ORU_R01|371784|P|2.5.1|||NE|NE|USA||||PHLabReportNoAck^ELR_Receiver^2.16.840.1.113883.9.11^ISO"),
+            Pair("Partially Terminated", "MSH|^~\\&"),
+            Pair("RANDOM TEXT", "foobar"),
+            Pair("NON-UTF", "®"),
+            Pair("EMOJI", "❤️ \uD83D\uDC94 \uD83D\uDC8C \uD83D\uDC95 \uD83D\uDC9E \uD83D\uDC93"),
+            Pair("LARGE POST BODY", a_60Meg_payload),
+        )
+
+        for (pair in badHl7Pairs) {
+            val (responseCode, jsonResponse) = HttpUtilities.postReportBytes(
+                environment,
+                (pair.second).toByteArray(),
+                sender,
+                options.key
+            )
+
+            TermUi.echo("ResponseCode to POST: $responseCode")
+
+            if (responseCode >= 400 && responseCode < 500) {
+                good("Test for $name ${pair.first} passed: received $responseCode response code.")
+            } else {
+                failures.add("${pair.first}")
+                bad("***Test for $name ${pair.first} FAILED***: Expected a failure HttpStatus***")
+                continue
+            }
+            try {
+                val tree = jacksonObjectMapper().readTree(jsonResponse)
+                val errorCount = tree["errorCount"].intValue()
+                if (errorCount > 0) {
+                    good("Test for $name ${pair.first} passed: ErrorCount of $errorCount was returned.")
+                } else {
+                    bad("***Test for $name ${pair.first} FAILED***: Expected a non-zero ErrorCount, got $errorCount error(s)")
+                    failures.add("${pair.first}")
+                    continue
```

Comment:
How about removing this continue, so the rest of the errors pop up too?

Reply:
done